### PR TITLE
Missing cats instances

### PR DESCRIPTION
**Pull request by oskin1**
`cats.Defer` and `cats.Alternative` instances for `Stream`.

### CODE REVIEW

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -4990,3 +4993,16 @@ private[fs2] trait StreamLowPriority {
         }
     }
 }
+
+private[fs2] trait StreamLowPriority1 {
```
Comment (kubukoz) [inline core/shared/src/main/scala/fs2/Stream.scala:4997]: I'm moderately sure I tried this already and it'll fail the laws :(

Comment (oskin1) [reply core/shared/src/main/scala/fs2/Stream.scala:4997]: @kubukoz, you mean there is no lawful `Alternative` instance for `Stream`?

Comment (SystemFw) [reply core/shared/src/main/scala/fs2/Stream.scala:4997]: See here https://github.com/functional-streams-for-scala/fs2/pull/1418

Comment (SystemFw) [reply core/shared/src/main/scala/fs2/Stream.scala:4997]: and discussion here https://github.com/functional-streams-for-scala/fs2/issues/1394

Comment (oskin1) [reply core/shared/src/main/scala/fs2/Stream.scala:4997]: I see, thanks!

**Comment by oskin1** [comment]
I'm closing this PR, going to create new one only adding `Defer` instance.

**Pull request closed without merging**